dev-proxy/ProxyEngine.cs

```diff
@@ -95,18 +99,16 @@ public async Task Run(CancellationToken? cancellationToken)
 
         var ipAddress = string.IsNullOrEmpty(_config.IPAddress) ? IPAddress.Any : IPAddress.Parse(_config.IPAddress);
         _explicitEndPoint = new ExplicitProxyEndPoint(ipAddress, _config.Port, true);
-        if (!RunTime.IsWindows)
+        // Fired when a CONNECT request is received
+        _explicitEndPoint.BeforeTunnelConnectRequest += OnBeforeTunnelConnectRequest;
+        if (_config.InstallCert)
         {
-            // we need to change this to a value lower than 397
-            // to avoid the ERR_CERT_VALIDITY_TOO_LONG error in Edge
-            _proxyServer.CertificateManager.CertificateValidDays = 365;
-            // we need to call it explicitly for non-Windows OSes because it's
-            // a part of the SetAsSystemHttpProxy that works only on Windows
             _proxyServer.CertificateManager.EnsureRootCertificate();
```

Comment:
We don't need to call this explicitly on Windows, because it's already taken care of as part of `_proxyServer.Start()`. Let's update the logic to reflect this

Reply:
If we only rely on `SetAsSystemHttpProxy`, it would break `--as-system-proxy false --install-cert true`.

Reply:
Ah, you're right. Carry on